.github/workflows/validate_plan_tf.yml

```diff
@@ -0,0 +1,152 @@
+name: TF Validate and Plan
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: Environment to terraform validate and plan (dev, stage, prod)
+        type: string
+        required: true
+      commit-identifier:
+        description: "SHA or tag to validate and plan terraform, (optional) when omitted, the head of the selected branch is used"
+        type: string
+        required: true 
+      with-lock:
+        description: whether to lock the state file or not, (default - true)
+        type: boolean
+        default: true
+      write-artifact:
+        description: write the terraform plan to an artifact
+        type: boolean
+        required: false
+        default: false
+      workspace: 
+        description: workspace to use
+        type: string
+        required: true
+    secrets:
+      ORG_READ_ONLY_SSH_KEY:
+        required: true
+      ORG_GITHUB_PACKAGES_READ_ONLY_TOKEN:
+        required: true
+    outputs:
+      plan-artifact-name:
+        description: the name of the plan artifact, if write-artifact is true
+        value: ${{ jobs.TF-Validate-Plan.outputs.plan-artifact-name}}
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ inputs.environment }}
+
+permissions:
+  id-token: write  # required by configure-aws-credentials
+  contents: read
+
+env:
+  TERRAFORM_ROOT: ${{ inputs.environment }}
```

Comment:
Where is this being used? I almost would have suggested that terraform root be an optional input and be defaulted based on environment if not supplied.

also default directory is more accurate to terraform_root

Reply:
not used at all, I'll just delete it